File: 15_FullStack/27.05.2024/hw_ExamPractice/hwexampractice_website/src/Components/Models/Task.ts
export class Task{
    public id:number;
    public name:string;
    public responsible:string;
    public dateScheduled: Date;
    public dateCompleted: Date;
    public isCompleted:boolean;

    constructor(id:number,name:string, responsible:string, dateScheduled:Date, dateCompleted:Date, isCompleted:boolean){
        this.id=id;
        this.name=name;
        this.responsible=responsible;
        this.dateScheduled=dateScheduled;
        this.dateCompleted=dateCompleted;
        this.isCompleted=isCompleted;
    }

}

File: 15_FullStack/27.05.2024/hw_ExamPractice/hwExamPractice_Spring/src/main/java/com/example/hwExamPractice_Spring/Beans/Task.java
package com.example.hwExamPractice_Spring.Beans;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Date;
import java.time.LocalDate;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    @Column(nullable = false)
    private String name;
    private String responsible;
    private LocalDate dateScheduled;
    @Column(nullable = true)
    private LocalDate dateCompleted;
    private boolean isCompleted;
}
